src/zarr/core/array.py

```diff
@@ -443,6 +449,55 @@ def basename(self) -> str | None:
             return self.name.split("/")[-1]
         return None
 
+    @property
+    @deprecated("AsyncArray.cdata_shape may be removed in an early zarr-python v3 release.")
+    def cdata_shape(self) -> ChunkCoords:
```

Comment:
I'm curious which of these helpers could migrate to the purview of the chunk grid.